git/working.go

```diff
@@ -170,11 +170,11 @@ func (c *Checkout) GetNote(ctx context.Context, rev string, note interface{}) (b
 }
 
 func (c *Checkout) HeadRevision(ctx context.Context) (string, error) {
-	return refRevision(ctx, c.dir, "HEAD")
+	return refRevision(ctx, c.dir, "heads/"+c.config.Branch)
```

Comment:
This changes the meaning of `HeadRevision` from "head of checked out branch" to "head of branch configured". I don't _think_ these can differ, as things stand, because there's no way to change the checked out branch of a clone. So I suppose this is a note to our future selves ..

Reply:
I would actually rename the function to make it more clear.

Reply:
I only changed it to create a clear usage pattern of `refRevision`.

Reverting the change to just lookup the `HEAD` will actually do us no harm as this will always be the revision of what is checked out at that moment.

Reply:
> Reverting the change to just lookup the `HEAD` will actually do us no harm as this will always be the revision of what is checked out at that moment.

I have a _slight_ preference for this, but it's pretty slight.

Reply:
I wouldn't totally revert it though, I guess a branch or a tag can also be named HEAD

Reply:
A branch can't be named `HEAD`, a tag can but `rev-list` will prioritize the real `HEAD` over the tag.